nodes.go

```diff
@@ -38,15 +38,25 @@ type node interface {
 	String() string
 }
 
-func decorate(err *error, name func() string) {
+func nearestPos(tokens []lexer.Token) *lexer.Position {
+	if len(tokens) == 0 {
+		return nil
+	}
+
+	return &tokens[0].Pos
+}
+
+func decorate(err *error, pos *lexer.Position, name func() string) {
 	if *err == nil {
 		return
 	}
-	if perr, ok := (*err).(Error); ok {
```

Comment:
It looks like this should
have already been preserving positional information to me, and that all that would need to be done is to ensure the error at the original call site had positional information.

Reply:
yes, it looks like this, but an arbitrary error returned from Capture is not a `participle.Error` so that code is never triggered.

Reply:
Similarly too with the `Parseable` interface.